bigquery/google/cloud/bigquery/client.py

```diff
@@ -1440,6 +1440,97 @@ def query(
 
         return query_job
 
+    def _convert_to_safe_json_dict(self, field, row_value):
+        """Maps a field and value to a JSON-safe value.
+
+        Args:
+            field ( \
+                :class:`~google.cloud.bigquery.schema.SchemaField`, \
+            ):
+                The SchemaField to use for type conversion and field name.
+
+            row_value (Union[ \
+                Sequence[list], \
+                any, \
+            ]):
+                Row value(s) to be converted. If the value is
+                a list, converted all elements in the list and return
+                a dictionary whose value is that converted list.
+
+        Returns:
+            dict:
+                A dict of a field name to value(s), encoded for JSON.
+        """
+        if not isinstance(row_value, list):
```

Comment:
We'd want to treat a tuple the same as we do a list. I'd prefer we look at `field.mode == 'REPEATED'` to decide this instead.

Can we refactor the "REPEATED" logic into a function `_repeated_field_to_json` instead? The remaining logic could be in a function named `_scalar_field_to_json`.